### PR TITLE
Project modals

### PR DESCRIPTION
**Pull request by AKeeganDev**
This pull request is intended to test a modal popup window that is generated when clicking either on a project card thumbnail OR a link to "see-project"

The modal window disappears when clicking the close icon generated on the top-right corner of the modal window.

The entire portfolio section and the modal window is generated dynamically through JS.

### CODE REVIEW

File: css/styles.css [hunk 1]
```diff
@@ -147,6 +191,32 @@ header {
...
+
+  @media screen and (max-width: 786px) {
+    button.modal-button {
+      font-size: 0.8rem;
+    }
+  }
+
+  .modal-details {
+    display: flex;
+    flex-direction: row;
```
Comment (micheaol) [inline css/styles.css:216]: Please ensure that the popup follows the design for both desktop and mobile versions from [the chosen template in Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1) (e.g., font, colours, images, tests, margins).
Your Implementation | Figma Design
:---:|:---:
image | image

File: js/script.js [hunk 2]
```diff
@@ -5,7 +5,145 @@ const barBottom = document.querySelector('.bar-bottom');
...
+  description: 'A daily selection of privately personalized reads; no accounts or sign-ups required. has been the industry\'s standard',
+  featuredIMGURL: null,
+  technologies: ['CSS', 'Bootsrap', 'Ruby'],
+  demoURL: '#',
+  linkToSource: '#',
+  button: null,
+};
+
+const cardArray = [card1, card2, card3, card4, card5, card6];
+
```
Comment (micheaol) [inline js/script.js:52]: [OPTIONAL]: Great implementation here. I think your code would have been more readable if you had implemented you `array of object` like this:

```javascript
const projects = [{
    name: "Submarine game",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
}]
```

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi @AKeeganDev , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: css/styles.css [hunk 3]
```diff
@@ -20,6 +20,21 @@ body {
...
+}
+
+p::-webkit-scrollbar {
+  background-color: #e4e4e4;
+  border-radius: 100px;
+  box-shadow: inset 0 0 5px grey;
+}
+
+p::-webkit-scrollbar-thumb {
+  background-color: #685c5c;
```
[the remark this answers is not in the document]
Comment (AKeeganDev) [reply css/styles.css:34]: @micheaol Thanks for the advice!

Question for you. What exactly do you want me to fix. I see the buttons can be spread out, but if you read the text (I know it is gibberish) the text is actually condensed on FIGMA and cuts off mid-sentence. I felt that for readability that it made sense to show that the text is cut off and to provide my user with a sidebar- unlike how it was done in FIGMA. I can either remove the sidebar and still allow for scrolling or I can cut the text off at the same place. Which would you prefer??

File: js/script.js (hunk 2 above)
Comment (AKeeganDev) [reply js/script.js:52]: @micheaol 

Thank you this is useful and I will consider adding it when I have the time!

**Review by keroloussamy — APPROVED** [review]
Hi @dewslyse, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️
- Add all the requirements. ✔️

clap



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AKeeganDev** [comment]
> Hi @dewslyse,
> 
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: Congratulations! tada
> 
> ## To highlight 
> * No linter errors.heavy_check_mark
> * Good code design and implementation. heavy_check_mark
> * Add all the requirements. heavy_check_mark
> 
> clap
> 
> Cheers and Happy coding!clapclapclap
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

@keroloussamy Thank you! This assignment was so difficult and time consuming that I literally want to cry in happiness now that this is approved hahahah 

Thank you! Have a great rest of the week! :)

**Pull request merged**